app/code/Magento/Swatches/view/frontend/web/js/swatch-renderer.js

```diff
@@ -840,27 +840,29 @@ define([
                 }
             );
 
-            if (result.oldPrice.amount !== result.finalPrice.amount) {
-                $(this.options.slyOldPriceSelector).show();
-            } else {
-                $(this.options.slyOldPriceSelector).hide();
-            }
+            if (typeof(result) != "undefined") {
```

Comment:
Though the `typeof` operator can be used with parentheses, it is not recommended, because it can confuse linters and reviewers. This should be `typeof result`.